Blog: Low-Code (#1771)
FlowFuse/website: src/blog/2024/03/low-code-is-better.md

<!--more-->

## My Coding Evolution

Every decade seems to introduce something that simplifies coding. For me, in college, it was Python. A classmate was excited about this new programming language that was all about simplicity and readability, especially with its indentation-based syntax. It seemed too simple at first. Yet, over time, Python became a staple in my programming toolbox.

During this time in our education, our coursework was filled with languages like C++, which felt distant from the future of programming we imagined. We joked about "outdated" languages, not yet realizing the breadth of what programming could encompass.

After finishing college in 2010, I started working during the tail end of the housing crisis in the U.S. My first role was as a Controls System Integrator at Logical System Inc., where I was introduced to programming PLCs with Ladder Logic. Despite my initial reservations—viewing it as barely a programming language—this experience was my first step towards appreciating the diversity and utility of programming languages beyond the conventional.

> Later on in my career, I worked as a Corporate Automation and Controls Engineer where I worked on and enforced standards for partner System Integrators and OEMs.  One of these important standards was making sure application written in the control process were written in Ladder Logic within the PLCs.  There were exceptions ofcourse, but the rule was to apply a visually appealing coding language, Ladder Logic, over a text like language often called Structured Text.

Author: ```suggestion
Later on in my career, I worked as a Corporate Automation and Controls Engineer where I worked on and enforced standards for partner System Integrators and OEMs.  One of these important standards was making sure applications written in the control process were written in Ladder Logic within the PLCs.  There were exceptions, of course, but the rule was to apply a visually appealing coding language, Ladder Logic, over a text-like language often called Structured Text.
```

---
title: Why Low-Code is Better
subtitle: Stop coding in High-Code when it can be done in Low-Code
description: When any programming language can solve a task, pick a low code tool where applicable.
date: 2024-03-11
authors: ["grey-dziuba"]
image: /blog/2024/03/images/low-code-is-better.png
tags:
    - posts
    - flowfuse
    - low-code
    - citizen development
---

> When I bring up the potential of low-code platforms shaping the future, I'm genuinely interested in sparking a conversation, not stirring debate for its own sake. I'm curious about your perspective. Do you see traditional coding as irreplaceable, or do you also believe in the shift towards low-code?

Author: Fair Working on it.

> <!--more-->

Author: I do agree.  Changed.

---
title: Why Low-Code is Better
subtitle: Stop coding in High-Code when it can be done in Low-Code
description: When any programming language can solve a task, pick a low code tool where applicable.
date: 2024-03-11
authors: ["grey-dziuba"]
image: /blog/2024/03/images/low-code-is-better.png
tags:
    - posts
    - flowfuse
    - low-code
    - citizen development
---

> A common reason why new languages come about is because of two reasons.  They provide a feature that is missing in the existing programming languages, or the other option is a tool that is easier to learn and use.  The latter often functions like a swiss army knife with each iteration including more and more tools.  The journey of low-code is like a swiss army knife, the perfect tool for the Citizen Developer.

Reviewer: ```suggestion
There are two common reasons why new languages come about.  They provide a feature missing in the existing programming languages, or it is a tool that is easier to learn and use.  The latter often functions like a swiss army knife with each iteration including more and more tools.  The journey of low-code is like a swiss army knife, the perfect tool for the Citizen Developer.
```

<!--more-->

> ## My Coding Evolution

Reviewer: ```suggestion
## A Typical Coding Evolution
```

The idea behind Citizen Development is simple, make programming accessible to more people. This is what low-code platforms aim to do. They lower the entry barrier, making programming more inclusive. Insisting on complex, traditional programming languages when there are simpler, equally powerful alternatives seems counterproductive. We should be looking towards making programming more accessible to everyone.

If we want to drive meaningful change within our organizations, embracing tools that broaden participation is key. The aim of adopting new standards and tools is to simplify, not complicate. It's about finding better, more accessible ways to work that can accommodate a wider range of skill sets.

## Simplification

This discussion isn't about the mechanics of coding, it's about opening up the field to more diverse contributions. Low-code platforms represent a step towards a more inclusive, collaborative future in technology. By lowering barriers to entry, we're not just simplifying coding, we're inviting a broader community to engage, innovate, and drive progress. I look forward to seeing how we can all contribute to this evolving landscape.

> ## FlowFuse

Reviewer: ```suggestion
## How FlowFuse Helps
```

> Our goal here at FlowFuse is to keep expanding on the swiss army knife, Node-RED. We strive to elevate Node-RED for professional by providing the tools needed to deploy Node-RED in a safe and secure way. We believe that Low-Code is the future and strive to make Citizen Development a reality.

Reviewer: ```suggestion
Our goal here at FlowFuse is to keep expanding on the Swiss army knife, Node-RED. We strive to elevate Node-RED for professionals by providing the tools needed to deploy Node-RED in a safe and secure way.  For example by default, the editor for Node-RED is protected using your FlowFuse user credentials. You can also use SSO to further protect your user accounts and give access to Node-RED to your team members. All traffic to FlowFuse and your Node-RED instances is protected by HTTPS. FlowFuse has set up the domain name and manages the certificates so you can spend time on your flows rather than configuring security.

We believe that Low-Code is the future and strive to make Citizen Development a reality. To learn more [schedule a call with one of our experts. ](https://flowfuse.com/free-consultation/)
```

---
title: Why Low-Code is Better
subtitle: Stop coding in High-Code when it can be done in Low-Code
description: When any programming language can solve a task, pick a low code tool where applicable.

> date: 2024-03-11

Author: ```suggestion
date: 2024-03-13
```
